### DISCUSSION TITLE
How to configure LSP syntax highlighting?

### DISCUSSION THREAD

**Discussion by barrelltech** [NvChad/NvChad#3096] [Q&A] [answered]
I just installed NVChad, and it looks incredible.

I added `clojure_lsp` to the lspconfig, installed it with Mason, and now when I open a clojure file I see it's enabled and I can navigate via LSP.

However, it doesn't seem to be effecting the formatting.

For example, `:hi verbose @lsp.type.macro` shows that they should print with red text, however they're printed in blue text like a normal function.

How can I enable LSP syntax highlighting? I've read through all the docs I can find and the answer is not clear to me

**Comment by barrelltech** [comment]
Ah ok so it appears the `on_init` default handler disables semanticTokens:

```lua
-- disable semanticTokens
M.on_init = function(client, _)
  if client.supports_method "textDocument/semanticTokens" then
    client.server_capabilities.semanticTokensProvider = nil
  end
end
```

So where I was doing:

```lua
-- load defaults i.e lua_lsp
require("nvchad.configs.lspconfig").defaults()

local lspconfig = require "lspconfig"

-- EXAMPLE
local servers = { "html", "cssls", "clojure_lsp" }
local nvlsp = require "nvchad.configs.lspconfig"

-- lsps with default config
for _, lsp in ipairs(servers) do
  lspconfig[lsp].setup {
    on_attach = nvlsp.on_attach,
    on_init = nvlsp.on_init,
    capabilities = nvlsp.capabilities,
  }
end
```

I now instantiate like:

```lua
-- lsps with default config
for _, lsp in ipairs(servers) do
  lspconfig[lsp].setup {
    on_attach = nvlsp.on_attach,
    capabilities = nvlsp.capabilities,
  }
end
```

And semantic tokens come back :)

**Comment by siduck** [comment]
thats right, also enable semantic_tokens base46 integration too! read :h nvui.base46

**Reply by barrelltech** [reply]
I found this:

```lua
Note: 

It's preferred to put the dofile code in your plugin's config func
>lua
 {
   "folke/trouble.nvim",
    cmd = "Trouble",
    config = function()
       dofile(vim.g.base46_cache .. "trouble")
       require("trouble").setup()
    end
 }
```

So I tried:

```lua
  {
    "neovim/nvim-lspconfig",
    config = function()
      dofile(vim.g.base46_cache .. "integrations/semantic_tokens")
      require "configs.lspconfig"
    end,
  },
```

And I got:

```lua
Failed to run `config` for nvim-lspconfig

cannot open /Users/myuser/.local/share/nvim/base46/integrations/semantic_tokens: No such file or directory

# stacktrace:
  - ~/.config/nvim/lua/plugins/init.lua:12 _in_ **config**
  - /NvChad/lua/nvchad/autocmds.lua:15
```

Fails with both `semantic_tokens` and `integrations/semantic_tokens`. What am I missing?

**Reply by siduck** [reply]
![image](https://github.com/user-attachments/assets/982e8cd5-25d5-4e5f-8cec-2751592a902d)

read!!!!!!!!!!!!!!!!!

**Reply by barrelltech** [reply, marked as answer]
I did. I'm a native English speaker who has been using Neovim for over a decade. I read through that multiple times and still had no idea what it's saying. I could have read it 100 times and still have no idea what to do.

Enough trial and error though and eventually I got it. To answer my own question, there are actually two steps to this. You need to install it, then require it.

The installation is the first part: in the `M.base64` object (found in `lua/chadrc.lua`), add the `integrations` object documented

The requirement is the second part: add the `doline` to the config param on the package requirement.

At least I think it's working. I see no difference, but I don't see any errors.

EDIT: Upon further use, adding the `semantic_tokens` integration breaks the semantic tokens. Removing the integration restored the proper functionality of semantic_tokens

**Reply by siduck** [reply]
how come you missed it?
```lua
M.base46 = { integrations =  { "semantic_tokens" } }
```

**Reply by guillenotfound** [reply]
I have the same feeling: https://github.com/NvChad/NvChad/discussions/3125

So if I got it right you have to add both an entry to integrations and then use the `dotfile` fn? My next question would be how can I figure it out if there's any missing integration that should be added? Because the colors will not look right for that particular thing?

**Reply by siduck** [reply]
@guillenotfound go check the default list of integrations in base46's repo's init.lua file. I hope you ran the dofile code somewhere. and make sure you have included your extra integration in M.base46.integrations field or else base46 wont include it for compiling

And do not use our on_init func  in your lspconfig if u want semantic coloring, cuz that func disables it
